src/main/java/org/aludratest/content/edifact/edifatto/EdiDiffDetailImpl.java

```diff
@@ -38,4 +38,19 @@ public EdiDiffDetailType getEdiDiffType() {
         return ediType;
     }
 
+    @Override
+    public int hashCode() {
+        return 31 * super.hashCode() + ((ediType == null) ? 0 : ediType.hashCode());
+    }
+
+    @Override
+    public boolean equals(Object obj) {
+        if (this == obj)
+            return true;
+        if (!super.equals(obj) || getClass() != obj.getClass())
```

Comment:
`super.equals` should never be true, as equality should always only match on SAME class (not instanceof). Please replace with `obj == null` check. Also, please add the curly braces here.